ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/BasicAction.java

```diff
@@ -2413,10 +2388,14 @@ protected void onePhaseCommit (boolean reportHeuristics)
                      */
                
                 if (p == TwoPhaseOutcome.ONE_PHASE_ERROR) {
-                   if (recordBeingHandled instanceof ExceptionDeferrer)
-                      onePhaseCommitExceptionDeferrer = (ExceptionDeferrer) recordBeingHandled;
-                   else if (recordBeingHandled.value() instanceof ExceptionDeferrer) 
-                      onePhaseCommitExceptionDeferrer = (ExceptionDeferrer) recordBeingHandled.value();
+                   if (recordBeingHandled instanceof ExceptionDeferrer) {
+                       ExceptionDeferrer onePhaseCommitExceptionDeferrer = (ExceptionDeferrer) recordBeingHandled;
+                       onePhaseCommitExceptionDeferrer.getDeferredThrowables(deferredThrowables);
+                   }
+                   else if (recordBeingHandled.value() instanceof ExceptionDeferrer) {
+                       ExceptionDeferrer onePhaseCommitExceptionDeferrer = (ExceptionDeferrer) recordBeingHandled.value();
+                       onePhaseCommitExceptionDeferrer.getDeferredThrowables(deferredThrowables);
+                   }
```

Comment:
Right I see now. Well, I didn't really add them, I just refactored something that was already there but yes to answer your question you can replace them.

I have gone even further and made the change as well